paimon-core/src/main/java/org/apache/paimon/utils/BranchManager.java

```diff
@@ -72,6 +73,9 @@ public Path branchPath(String branchName) {
     }
 
     public void createBranch(String branchName, String tagName) {
+        checkArgument(
+                !branchName.equals(DEFAULT_MAIN_BRANCH),
+                "Branch name 'main' is the default branch and cannot be used.");
         checkArgument(!StringUtils.isBlank(branchName), "Branch name '%s' is blank.", branchName);
```

Comment:
Use `DEFAULT_MAIN_BRANCH ` instead of 'main' here.

Reply:
I have fixed it.